src/apps/Permissions/PermissionsView.js

```diff
@@ -0,0 +1,170 @@
+import React, { useCallback, useMemo } from 'react'
+import PropTypes from 'prop-types'
+import {
+  ButtonIcon,
+  ContextMenu,
+  ContextMenuItem,
+  DataView,
+  IconTrash,
+  textStyle,
+  useLayout,
+  useTheme,
+} from '@aragon/ui'
+import { usePermissions } from '../../contexts/PermissionsContext'
+import LocalIdentityBadge from '../../components/IdentityBadge/LocalIdentityBadge'
+import PermissionsIdentityBadge from './PermissionsIdentityBadge'
+
+function PermissionsView({ permissions, onManageRole, heading, showApps }) {
+  const { layoutName } = useLayout()
+  const willRenderEntryChild = useMemo(
+    () => permissions.some(permission => permission.entities.length > 1),
+    [permissions]
+  )
+
+  const fields = [
+    'Action',
+    'On app',
+    { label: 'Assigned to entity', childStart: true },
+    'Managed by',
+  ]
+
+  if (!showApps) {
+    fields.splice(1, 1)
+  }
+
+  return (
+    <DataView
+      heading={heading}
+      mode={
+        layoutName === 'large' || (layoutName === 'medium' && !showApps)
+          ? 'table'
+          : 'list'
+      }
+      fields={permissions.length ? fields : []}
+      entries={permissions}
+      renderEntry={entry => renderEntry(entry, showApps)}
```

Comment:
Is this going to send a new function every time and trigger re-render?

Reply:
Maybe we could figure a way to send extra params to each entry on the render function? To have `renderEntry={renderEntry}` and render entry still gets the `showApps` param

Reply:
Yes we could memoize it, but I opted for memoizing `PermissionsView` directly, instead of the `DataView` props. That way, it (and `DataView` below) renders only when needed.

Reply:
Yeah, you solved this, but maybe adding the improvement to `DataView` helps dev consumers?

Reply:
It’s going to be hard to do from `DataView`: if a function reference gets updated, we don’t know if it’s intended or not… like other React components, users should be aware that an inline function will trigger a rerender.

But I am thinking of memoizing `DataView`, so that at least it won’t rerender when the parent updates, as long as the passed props are the same.